### PR TITLE
Feature/6623 UI tests and screenshots

### PR DESCRIPTION
**Pull request by freshking**
## Description

UI tests for antigen test profile

## Link to Jira

https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6623

### CODE REVIEW

File: src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITestsAntigenTestProfile.swift [hunk 1]
```diff
@@ -0,0 +1,141 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import XCTest
+
+class ENAUITestsAntigenTestProfile: XCTestCase {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITestsAntigenTestProfile.swift:7]: Please move this to ENAUITests folder, check for the existing test file (where PCR tests are) if there is already, if not please name it like "ENAUITests_12_AntigenTestProfile"

**Review by naveeddotio — DISMISSED** [review]
Please move UITest file to ENAUITests folder, check for the existing test file (where PCR tests are) if there are already, if not please name it like "ENAUITests_12_AntigenTestProfile for consistency"

**Review by KaiTeuber — APPROVED** [review]

File: src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+	}
+	
+	// MARK: - Internal
+
+	var app: XCUIApplication!
+	
+	// MARK: - Tests
+	
+	func test_FIRST_CreateAntigenTestProfile_THEN_DeleteProfile() throws {
+
```
Comment (naveeddotio) [inline src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift:29]: Do you think it make sense to capture two/three snapshots during testing?

File: src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+	// MARK: - Tests
+	
+	func test_FIRST_CreateAntigenTestProfile_THEN_DeleteProfile() throws {
+		
+		app.launch()
+		
+		/// Home Screen
+
+		let registerTestButton = try XCTUnwrap(app.buttons[AccessibilityIdentifiers.Home.submitCardButton])
+		registerTestButton.tap()
```
Comment (pascal-brause) [inline src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift:35]: add here before the tap `registerTestButton.waitForExistence()`

File: src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+		
+		let dataPrivacyButton = try XCTUnwrap(app.cells[AccessibilityIdentifiers.ExposureSubmission.AntigenTest.Information.dataPrivacyTitle])
+		dataPrivacyButton.tap()
+		
+		/// Legal Text Screen
+		
+		XCTAssertTrue(app.cells[AccessibilityIdentifiers.ExposureSubmission.AntigenTest.Information.dataPrivacyTitle].exists)
+		
+		let backButton = try XCTUnwrap(app.navigationBars.buttons.element(boundBy: 0))
+		backButton.tap()
```
Comment (pascal-brause) [inline src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift:62]: add here before the tap `backButton.waitForExistence()`

File: src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+		cityTextField.tap()
+		cityTextField.typeText("München")
+		
+		let phoneNumberTextField = try XCTUnwrap(app.cells.textFields[AccessibilityIdentifiers.AntigenProfile.Create.phoneNumberTextField])
+		phoneNumberTextField.tap()
+		phoneNumberTextField.typeText("089123456")
+		
+		let emailAddressTextField = try XCTUnwrap(app.cells.textFields[AccessibilityIdentifiers.AntigenProfile.Create.emailAddressTextField])
+		emailAddressTextField.tap()
+		emailAddressTextField.typeText("bastian@bastian.codes")
```
Comment (pascal-brause) [inline src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift:105]: for every tab, you should check with .exists or .waitForExistence()

File: src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+		saveProfileButton.tap()
+		
+		/// Antigen Test Profile Screen
+
+		// continues button exists
+		XCTAssertTrue(app.buttons[AccessibilityIdentifiers.ExposureSubmission.AntigenTest.Profile.continueButton].exists)
+		// delete profile button exists
+		XCTAssertTrue(app.buttons[AccessibilityIdentifiers.ExposureSubmission.AntigenTest.Profile.deleteButton].exists)
+		
+		let closeButton = try XCTUnwrap(app.navigationBars.buttons.element(boundBy: 1))
```
Comment (pascal-brause) [inline src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift:117]: again see above

File: src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+		
+		/// -> Home Screen
+
+		registerTestButton.tap()
+		
+		/// Register Test Screen
+		
+		// test profile button exists
+		let testProfileButton = try XCTUnwrap(app.buttons[AccessibilityIdentifiers.ExposureSubmission.AntigenTest.Profile.profileTile_Description])
+		testProfileButton.tap()
```
Comment (pascal-brause) [inline src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift:128]: again see above

File: src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+		XCTAssertTrue(app.buttons[AccessibilityIdentifiers.ExposureSubmission.AntigenTest.Profile.continueButton].exists)
+		// delete profile button exists
+		XCTAssertTrue(app.buttons[AccessibilityIdentifiers.ExposureSubmission.AntigenTest.Profile.deleteButton].exists)
+		
+		let closeButton = try XCTUnwrap(app.navigationBars.buttons.element(boundBy: 1))
+		closeButton.tap()
+		
+		/// -> Home Screen
+
+		registerTestButton.tap()
```
Comment (pascal-brause) [inline src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift:122]: again see above

File: src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift (hunk 3 above)
Comment (freshking) [reply src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift:35]: Why do we need this?

File: src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift (hunk 4 above)
Comment (freshking) [reply src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift:62]: Why do we need this?

File: src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift (hunk 5 above)
Comment (freshking) [reply src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift:105]: Why do we need this?

File: src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift (hunk 6 above)
Comment (freshking) [reply src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift:117]: Why do we need this?

File: src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift (hunk 7 above)
Comment (freshking) [reply src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift:128]: Why do we need this?

File: src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift (hunk 8 above)
Comment (freshking) [reply src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift:122]: Why do we need this?

File: src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift (hunk 2 above)
Comment (freshking) [reply src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift:29]: I have no idea. Nothing was specified in the story.

File: src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift (hunk 3 above)
Comment (naveeddotio) [reply src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift:35]: To make sure if the button which you want to tab is there already, reduces the flakiness of the UI tests

**Review by pascal-brause — APPROVED** [review]
❤️

File: src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate.swift [hunk 9]
```diff
@@ -651,6 +652,17 @@ class AppDelegate: UIResponder, UIApplicationDelegate, CoronaWarnAppDelegate, Re
...
 
+	private func setupAntigenTestProfileForTesting() {
+		if let antigenTestProfileInfoScreenShown = UserDefaults.standard.string(forKey: "antigenTestProfileInfoScreenShown") {
+			store.antigenTestProfileInfoScreenShown = antigenTestProfileInfoScreenShown != "NO"
+		}
+		if let removeAntigenTestProfile = UserDefaults.standard.string(forKey: "removeAntigenTestProfile") {
+			if removeAntigenTestProfile == "YES" {
+				store.antigenTestProfile = nil
+			}
+		}
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate.swift:663]: why no bool?
```suggestion
		if UserDefaults.standard.bool(forKey: "antigenTestProfileInfoScreenShown") {
			store.antigenTestProfileInfoScreenShown.setValue(false, forKey: "antigenTestProfileInfoScreenShown")
		}
		if UserDefaults.standard.bool(forKey: "removeAntigenTestProfile") {
			store.antigenTestProfile = nil
		}
```

**Review by whiskey — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate.swift (hunk 9 above)
Comment (freshking) [reply src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate.swift:663]: I copied code from other implementations. Can you pass a bool value to the launch arguments?

Comment (naveeddotio) [reply src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate.swift:663]: At the moment no, so having strings are fine from my POV.

**Review by naveeddotio — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate.swift (hunk 9 above)
Comment (freshking) [reply src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate.swift:663]: @whiskey I seems we have to do it this way.

Comment (whiskey) [reply src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate.swift:663]: @freshking @naveeddotio You CAN!

Screen Shot 2021-05-07 at 12 16 25
Screen Shot 2021-05-07 at 12 16 41
(tested with my app which doesn't take ages to start)

Comment (naveeddotio) [reply src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate.swift:663]: Honestly, I tried this with Bool and it didn't work for me. So, I looked in to the app and it was string everywhere, so I changed it to string. If it works for you Bastian, give it a try.

Comment (whiskey) [reply src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate.swift:663]: Have you tried the "good" old Obj-C bools? Not that fancy new Swift shit you young kids play the whole time! 👨🏻‍🦳

😉

**Review by whiskey — APPROVED** [review]

File: src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift [hunk 10]
```diff
@@ -13,10 +13,10 @@ class ENAUITests_12_AntigenTestProfile: XCTestCase {
...
 		app = XCUIApplication()
 		app.setDefaults()
-		app.launchArguments.append(contentsOf: ["-isOnboarded", "YES"])
-		app.launchArguments.append(contentsOf: ["-setCurrentOnboardingVersion", "YES"])
-		app.launchArguments.append(contentsOf: ["-antigenTestProfileInfoScreenShown", "NO"])
-		app.launchArguments.append(contentsOf: ["-removeAntigenTestProfile", "YES"])
+		app.launchArguments.append(contentsOf: ["-isOnboarded", YES])
+		app.launchArguments.append(contentsOf: ["-setCurrentOnboardingVersion", YES])
+		app.launchArguments.append(contentsOf: ["-antigenTestProfileInfoScreenShown", NO])
+		app.launchArguments.append(contentsOf: ["-removeAntigenTestProfile", YES])
```
Comment (whiskey) [inline src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift:19]: Unnecessary move but now it's in 🤷‍♂️

Comment (freshking) [reply src/xcode/ENA/ENAUITests/AntigenTestProfile/ENAUITests_12_AntigenTestProfile.swift:19]: Why is this unnecessary? I need to set the info screen to not shown so I can test it.

**Pull request merged**